app/models/comment.rb

```diff
@@ -0,0 +1,8 @@
+class Comment < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: :author_id
+  belongs_to :post
+
+  def update_post_comments_counter
+    post.update(comments_counter: post.comments.count)
+  end
```

Comment:
- [ ] [OPTIONAL]
Similarly, it's a good practice to use one of the Rails Active Callbacks `after_save` to call the `update_comment_counter` method every time a new comment has been saved to the database, and when `after_save` is called mark the `update_comment_counter` method as _private_ accordingly, which should be similar to:

```ruby
 # app/models/comment.rb
  after_save :update_comment_counter

  private

  def update_comment_counter
    author.increment!(:comments_counter)
  end
```

Doing so will enable the triggering of update logic at the right instance.